### DISCUSSION TITLE
Limitations in shared memory transport

### DISCUSSION THREAD

**Discussion by tmayoff** [OpenDDS/OpenDDS#4088] [Q&A]
I'm experiencing a weird issue when using the shared memory transport, I don't have an isolated example as of yet but occasionally when too many things are being written to the shared memory transport everything stops. OpenDDS logs are still logging about the data being sent seemingly without issue but nothings is being received.

```ini
[common]
DCPSDebugLevel=5
DCPSMonitor=0

[domain/1]
DefaultTransportConfig=local_config
DiscoveryConfig=DEFAULT_RTPS

[config/local_config]
transports=sharedmem

[transport/sharedmem]
transport_type=shmem
; pool_size=201326592
; datalink_control_size=16384
```

I've messed around with the config options in OpenDDS for the transport however nothing seems to change. Are there known limitations to the shared memory transport, or is it something worth investigating more? I'm able to fix/break the transport when I comment/uncomment the write lines I have for one of the writers I have

**Comment by iguessthislldo** [comment]
There were problems with shmem on start up relatively recently (#3549 and #4028), so it might be a new aspect of those issues if it's happening when new datawriters and datareaders are joining. #2799 sounds like it might be related with [everything stopping](https://github.com/OpenDDS/OpenDDS/issues/2799#issuecomment-1308937940), though it looks like you already tried increasing `datalink_control_size`.
